skbio/stats/power.py

```diff
@@ -476,44 +485,34 @@ def subsample_paired_power(test, meta, cat, control_cats, order=None,
 
-    # Determines the minimum number of ids avaliable
-    num_ids = len(sub_ids[0])
-
-    # Checks there are enough samples to subsample
-    if num_ids <= min_observations:
-        raise ValueError('There are not enough samples for subsampling.')
-
-    # Calculates the effect size vector
-    if min_counts is None:
-        min_counts = counts_interval
-
-    if (max_counts - min_counts) < counts_interval:
-        raise ValueError("No subsamples of the specified size can be drawn.")
-
-    sample_counts = np.arange(min_counts,
-                              min(max_counts, num_ids),
-                              counts_interval)
+    ratio, num_p, sample_counts = \
+        _check_subsample_power_inputs(test=test,
+                                      samples=sub_ids,
+                                      draw_mode='matched',
+                                      min_observations=min_observations,
+                                      min_counts=min_counts,
+                                      max_counts=max_counts,
+                                      counts_interval=counts_interval)
 
     # Prealocates the power array
-    power = np.zeros((num_runs, len(sample_counts)))
-
-    power[0, :] = _calculate_power_curve(test,
-                                         sub_ids,
-                                         sample_counts,
-                                         mode="matched",
-                                         num_iter=num_iter,
-                                         alpha=alpha_pwr)
-
-    for id1 in np.arange(1, num_runs):
-        sub_ids = paired_subsamples(meta, cat, control_cats, order,
-                                    strict_match)
-        # Calculates the power curve
-        power[id1, :] = _calculate_power_curve(test,
-                                               sub_ids,
-                                               sample_counts,
-                                               num_iter=num_iter,
-                                               alpha=alpha_pwr,
-                                               mode="matched")
+    power = np.zeros((num_runs, len(sample_counts), num_p))
+
+    # Calculates power instances
+    for id2, c in enumerate(sample_counts):
+        count = np.round(c * ratio, 0).astype(int)
+        for id1 in range(num_runs):
+            sub_ids = paired_subsamples(meta, cat, control_cats, order,
+                                        strict_match)
+            ps = _compare_distributions(test=test,
+                                        samples=sub_ids,
+                                        num_p=num_p,
+                                        counts=count,
+                                        num_iter=num_iter,
+                                        mode="matched")
+            power[id1, id2, :] = _calculate_power(ps, alpha_pwr)
+
+    if num_p == 1:
+        power = power[:, :, 0]
 
     return power, sample_counts
```

Comment:
Exactly.  But especially looking at `power` it isn't immediately obvious which variables are responsible for which dimension until I look at the source code.